src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int newCapacity = 0;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = 0;
+        if (key != null) {
+            index = Math.abs(key.hashCode()) % table.length;
+        }
+        while (true) {
+            if (table[index] == null) {
+                table[index] = new Node<>(objectHash(key), key, value, null);
+                size++;
+                if (size >= threshold) {
+                    resize();
+                }
+                return;
+            } else if (areValuesEqual(table[index].key, key)) {
+                table[index].value = value;
+                return;
+            } else {
+                index = (index + 1) % table.length;
```

Comment:
if I got right what you have here is you do not use your linked list in each bucket what you are doing instead is jumping to the next bucket where you should have iterated through the nodes that are not in the array nut in a single-linked list, we should fix that